### PR TITLE
[PE-4033] Combine components javascript and custom elements into a single view

### PR DESCRIPTION
**Pull request by janibolkvadze**
### Initial commit summary

- Restructured all the components
Example:
`Alert`
`  |__ _accessibility.pug`
`  |__ _examples.pug`
`  |__ _properties.pug`
`  |__ index.pug`
Leading underscore is needed to hide the pug file from gulp nav list generator.
The index file is index.pug (not alert.pug) because this way we avoid duplication of component name in the link `/components/alert/alert` -> `/components/alert` and the URLs stay the same as are, which is good for SEO reasons.
- Defined descriptions for all the components (To display them under component title and generate meta descriptions)
- Added feature of generating view tabs (Examples, Properties, Accessibility, etc) dinamically like:
```
---
title: Alert
description: Alert messages provide contextual feedback for user actions.
tabs: {"Examples" : "examplestab", "Properties" : "propertiestab", "Accessibility" : "accessibilitytab"}
---
```
- Implemented new design

### Current progress

| --- | Examples | Texts | Properties | Review |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| Alert  | **DONE**  | **DONE** | **DONE** | __ |
| Badge | **DONE** | **DONE** | **DONE** | __ |
| Brand | **DONE** | **DONE** | **DONE** | __ |
| Button | **DONE** | **DONE** | **DONE** | __ |
| Date picker  | **DONE**  | **DONE** | **DONE** | __ |
| Drawer  | **DONE**  | **DONE** | **DONE** | __ |
| Expansion panel  | **DONE**  | **DONE** | **DONE** | __ |
| Icon  | **DONE**  | **DONE** | **DONE** | __ |
| Number input  | **DONE**  | **DONE** | **DONE** | __ |
| Popover  | **DONE**  | **DONE** | **DONE**  | __ |
| Progress  | **DONE**  | **DONE** | **DONE** | __ |
| Spinner  | **DONE**  | **DONE** | **DONE** | __ |
| Toggle switch  | **DONE**  | **DONE** | **DONE** | __ |
| Dropdown  | **DONE**  | **DONE** | **DONE** | __ |
| Modal  | **DONE**  | **DONE** | **DONE** | __ |
| Range slider  | **DONE**  | **DONE** | **DONE** | __ |
| Sidenav  | **DONE**  | **DONE** | **DONE** | __ |
| Tabs  | **DONE**  | **DONE** | **DONE** | __ |

@mattnickles @jllr 

https://ctl.atlassian.net/browse/PE-4033

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/485/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/486/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/489/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/505/ for more details.

File: src/website/assets/styles/_docs-container.scss [hunk 1]
```diff
@@ -86,21 +85,33 @@
...
 
-        >.a-grid__container {
+
+        .a-grid__container {
           @include respond-to(lg) {
             margin: 0 40px;
           }
 
           @include respond-to(xl) {
             margin: 0 auto;
```
[the remark this answers is not in the document]
Comment (janibolkvadze) [reply src/website/assets/styles/_docs-container.scss]: Removing margin breaks center alignment. I added `padding: 0 20px;` to fix responsive behavior.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/507/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/509/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/510/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/511/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/512/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/513/ for more details.

**Comment by DiegoSanjuan** [comment]
@janibolkvadze it looks like 5 days ago node update docker images and broke wget binary path.
https://github.com/docker-library/repo-info/blob/master/repos/node/remote/dubnium-stretch-slim.md
So os we use another docker image, a fixed version older one, or the PRs will not pass, as they tried to use wget for downloading.

Let me know if you need me for fix this.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/515/ for more details.

**Comment by DiegoSanjuan** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/520/ for more details.

**Comment by janibolkvadze** [comment]
> test this please

@DiegoSanjuan Thanks for the fix!

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/527/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/550/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/554/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/555/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/556/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/557/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/561/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/593/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/604/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/606/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/607/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/609/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/614/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/615/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/616/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/617/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/618/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/619/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/620/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/621/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/622/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/623/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/624/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/625/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/626/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/627/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/628/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/629/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://c03682e9a8055eea63ec9d14dcb3be6bf5e3d4be-chi-pr.enterprise.ctl.io" target="_blank"> https://c03682e9a8055eea63ec9d14dcb3be6bf5e3d4be-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/662/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://012ec75592c6aad7e4fab562744ddc86c184ff49-chi-pr.enterprise.ctl.io" target="_blank"> https://012ec75592c6aad7e4fab562744ddc86c184ff49-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/686/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://8033cd27eb2557f488bd1396af540619f7a96ad1-chi-pr.enterprise.ctl.io" target="_blank"> https://8033cd27eb2557f488bd1396af540619f7a96ad1-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/689/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://e3a811c960dfd5a6a19549ec42b0a780c0bf26e8-chi-pr.enterprise.ctl.io" target="_blank"> https://e3a811c960dfd5a6a19549ec42b0a780c0bf26e8-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/691/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://53dd0cfd7099e0ae4bfd6d5cb848759b517ffb34-chi-pr.enterprise.ctl.io" target="_blank"> https://53dd0cfd7099e0ae4bfd6d5cb848759b517ffb34-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/693/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://fcc56245d4b4d30ca985333fe6bb565e1a41e5de-chi-pr.enterprise.ctl.io" target="_blank"> https://fcc56245d4b4d30ca985333fe6bb565e1a41e5de-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/694/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://38ddf490943dd02530f990ec51a89d2694f07b2b-chi-pr.enterprise.ctl.io" target="_blank"> https://38ddf490943dd02530f990ec51a89d2694f07b2b-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/695/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://5b5cfd689fd64c1b5128fca0696f6c8ed7d90a66-chi-pr.enterprise.ctl.io" target="_blank"> https://5b5cfd689fd64c1b5128fca0696f6c8ed7d90a66-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/698/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-l0bbyc, https://ops.elasticbox.com/services/instances/i-mk0gc2, https://ops.elasticbox.com/services/instances/i-uvplfh, https://ops.elasticbox.com/services/instances/i-addzrw, https://ops.elasticbox.com/services/instances/i-c6g3r7, https://ops.elasticbox.com/services/instances/i-61uvrv, https://ops.elasticbox.com/services/instances/i-f2fc1f, https://ops.elasticbox.com/services/instances/i-p6hkrk

**Pull request merged**